### PR TITLE
Combining AMA and legacy tasks

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects #6617

### Description
This PR makes a few changes:
1) Replaces AmaTask and LegacyTask with Task in Flow.
2) Translates the Task response from the backend into a more javascript-y structure (no `.attributes` and use camelCase)
3) Consolidates the AmaTaskTable with TaskTable.
4) Task tables now show both a users AMA and legacy tasks.

### Testing Plan
1. A manual check of all the task tables shows no regressions.
1. Going to a judge queue with AMA tasks (BVAAABSHIRE) shows AMA and legacy appeals.

### CODE REVIEW

File: app/models/serializers/work_queue/task_serializer.rb [hunk 1]
```diff
@@ -24,19 +24,23 @@ class WorkQueue::TaskSerializer < ActiveModel::Serializer
     object.appeal.docket_number
   end
 
-  attribute :veteran_name do
+  attribute :veteran_full_name do
```
Comment (mdbenjam) [inline app/models/serializers/work_queue/task_serializer.rb]: Bringing this in line with the legacy_task_serializer

File: client/app/queue/utils.js [hunk 2]
```diff
@@ -23,34 +21,30 @@ import DIAGNOSTIC_CODE_DESCRIPTIONS from '../../constants/DIAGNOSTIC_CODE_DESCRI
...
-export const prepareTasksForStore = (tasks: Array<Object>): AmaTasks => tasks.reduce((acc, curr: AmaTask) => {
-  acc[curr.attributes.external_id] = curr;
-
-  return acc;
-}, {});
-
-export const prepareLegacyTasksForStore = (tasks: Array<Object>): LegacyTasks => {
-  const mappedLegacyTasks = tasks.map((task) => {
-    return {
+export const prepareTasksForStore = (tasks: Array<Object>): Tasks => {
```
Comment (dannysellers) [inline client/app/queue/utils.js]: It's out of scope for this PR, but I feel like we should move these prepare-for-store methods to a dedicated file / come up with a name for this type of method (they're not quite factories, and are different than dispatch actions, I dunno 🤷‍♂️)

Comment (mdbenjam) [reply client/app/queue/utils.js]: I definitely agree with moving this out of utils 👍

File: client/app/queue/selectors.js [hunk 3]
```diff
@@ -30,31 +29,21 @@ const getAppealId = (state: State, props: Object) => props.appealId;
...
-      return { ...appeal,
-        tasks: _.filter(tasks, (task) => task.externalAppealId === appeal.externalId) };
-    });
+export const tasksWithAppealSelector = createSelector(
+  [getTasks, getAmaTasks, getAppeals],
+  (tasks: Tasks, amaTasks: Tasks, appeals: Appeals) : Array<TaskWithAppeal> => {
+    return [
+      ..._.map(tasks, (task) => {
+        return { ...task,
+          appeal: _.find(appeals, (appeal) => task.externalAppealId === appeal.externalId)
```
[the remark this answers is not in the document]
Comment (mdbenjam) [reply client/app/queue/selectors.js]: Let's chat about this when I get back. Might help us organize our code.

File: client/app/queue/components/TaskTable.jsx [hunk 4]
```diff
@@ -56,45 +57,21 @@ class TaskTable extends React.PureComponent<Props> {
     return isTaskSelected[externalAppealId] || false;
   }
 
-  appealHasDASRecord = (appeal: BasicAppeal) => {
-    if (this.props.requireDasRecord && appeal.tasks) {
-      return appeal.tasks.some((task) => task.taskId);
+  taskHasDASRecord = (task: TaskWithAppeal) => {
+    if (this.props.requireDasRecord) {
+      return task.appeal.docketName !== 'Legacy';
```
[the remark this answers is not in the document]
Comment (mdbenjam) [reply client/app/queue/components/TaskTable.jsx]: Good catch, I think I fixed this.

**Pull request merged**